src/mount-support.c

```diff
@@ -482,20 +558,66 @@ void sc_populate_mount_ns(const char *security_tag)
 	if (vanilla_cwd == NULL) {
 		die("cannot get the current working directory");
 	}
-	// Make our "/" a rslave of the real "/". This means that mounts from the
-	// host "/" get propagated to our namespace (i.e. we see new media mounts).
-	if (mount("none", "/", NULL, MS_REC | MS_SLAVE, NULL) != 0) {
-		die("can not make make / rslave");
-	}
+	// Remeber if we are on classic, some things behave differently there.
 	bool on_classic = is_running_on_classic_distribution();
-	// do the mounting if run on a non-native snappy system
 	if (on_classic) {
-		setup_snappy_os_mounts();
+		const char *bidirectional_mounts[] = {
+#ifdef MERGED_USR
+			"/run/media",	// access to the users removable devices
+#else
+			"/media",	// access to the users removable devices
+#endif				// MERGED_USR
+			NULL,
+		};
+		const char *unidirectional_mounts[] = {
+			"/dev",	// because it contains devices on host OS
+			"/etc",	// because that's where /etc/resolv.conf lives, perhaps a bad idea
+			"/home",	// to support /home/*/snap and home interface
+			"/root",	// because that is $HOME for services
+			"/proc",	// fundamental filesystem
+			"/sys",	// fundamental filesystem
+			"/tmp",	// to get writable tmp
+			"/var/snap",	// to get access to global snap data
+			"/var/lib/snapd",	// to get access to snapd state and seccomp profiles
+			"/var/tmp",	// to get access to the other temporary directory
+			"/run",	// to get /run with sockets and what not
```

Comment:
Ah, correct. I was not testing this on fedora (no fedora CI yet) and that's where we use merged user. Nice catch!